### ISSUE TITLE
Python3 map2alm ring weights string conversion error

### ISSUE THREAD

**Issue by cjcopi** [healpy/healpy#363]
map2alm fails to read the ring weights due to a string conversion error.  This is presumably the "usual" Python3 string versus bytes issue, though I have not looked into the details.  map2alm works as expected when run with Python2.

Explicitly, when using
`map2alm(mp, use_weights=True)`
I get (here shown run under a recent anaconda install, I get the same result using a Python 3.4.2 install)
```
TypeError                                 Traceback (most recent call last)
<ipython-input-8-a85dcb21f7a3> in <module>()
----> 1 hp.map2alm(mp,use_weights=True)

/home/copi/tmp/anaconda3/lib/python3.5/site-packages/healpy/sphtfunc.py in map2alm(maps, lmax, mmax, iter, pol, use_weights, datapath)
    155         alms = _sphtools.map2alm(maps, niter = iter, 
    156                                  datapath = datapath, use_weights = use_weights,
--> 157                                  lmax = lmax, mmax = mmax)
    158     else:
    159         # info >= 2 and pol is False : spin 0 spht for each map

_sphtools.pyx in healpy._sphtools.map2alm (healpy/src/_sphtools.cpp:4308)()

TypeError: expected bytes, str found
```

If instead the datapath is explicitly forced to be bytes as in
`map2alm(mp, use_weights=True, datapath=b'/path/to/ring/file/dir')`
I get the error
```
/home/copi/tmp/anaconda3/lib/python3.5/site-packages/healpy/sphtfunc.py in map2alm(maps, lmax, mmax, iter, pol, use_weights, datapath)
    155         alms = _sphtools.map2alm(maps, niter = iter, 
    156                                  datapath = datapath, use_weights = use_weights,
--> 157                                  lmax = lmax, mmax = mmax)
    158     else:
    159         # info >= 2 and pol is False : spin 0 spht for each map

_sphtools.pyx in healpy._sphtools.map2alm (healpy/src/_sphtools.cpp:4357)()

/home/copi/tmp/anaconda3/lib/python3.5/posixpath.py in join(a, *p)
     87                 path += sep + b
     88     except (TypeError, AttributeError, BytesWarning):
---> 89         genericpath._check_arg_types('join', a, *p)
     90         raise
     91     return path

/home/copi/tmp/anaconda3/lib/python3.5/genericpath.py in _check_arg_types(funcname, *args)
    143                             (funcname, s.__class__.__name__)) from None
    144     if hasstr and hasbytes:
--> 145         raise TypeError("Can't mix strings and bytes in path components") from None

TypeError: Can't mix strings and bytes in path components
```

Finally, if I instead read the weights into an array and pass them in, it works as expected.  Thus it is "just" a string conversion issue.

The string conversion could also be of importance for issue #356

**Comment by zonca** [comment]
thanks for reporting this, would you be able to provide a Pull Request to fix this?

**Comment by zonca** [comment]
@cjcopi would you be able to provide a fix? Thanks

**Comment by zonca** [comment]
fixed in #369
